optical/converter/tfrecord.py

```diff
@@ -0,0 +1,125 @@
+"""
+__author__: HashTagML
+license: MIT
+Created: Friday, 16th April 2021
+"""
+
+import os
+import io
+import warnings
+from typing import Union
+from pathlib import Path
+
+from PIL import Image
+import numpy as np
+import pandas as pd
+
+try:
+    import tensorflow as tf
+except Exception:
+    tf = None
+
```

Comment:
and here! also, I don't think we need the check in every place (i may be wrong though). I guess we can have the check in `Annotation` - `tfrecords.py` is not hit if someone doesn't request "to" or "from" `tfrecord`. As such, we can see if TF is installed in `__init__` and raise there itself.

Please see if that works. Seems a better design.

Reply:
and just import `tensorflow` inside the functions as needed.

Reply:
If there is no compelling reason to import inside a function we should avoid that. Though there is controversy in this subject, PEP 8 specifies all the imports should be  essentially done at the top.

In this case we can try to set the flag at top and check that flag inside the function. Seems a better design to me with better maintainability.

Reply:
I agree, but here the case is, TensorFlow is being imported in multiple places like `base.py`, `tfrecord.py`, and `utils.py` for various functions. So while calling "to" and "from" `tfrecord` format, he has to have this import. how can we make sure that tensforflow is imported only when `tfrecord` is used?

Reply:
It's a reasonable consideration. To that, my thought is we can place this "gate" (check for tf installation) at the two entrypoints - 
1. in `converter.__init__.py` for "from"
2. in `converter.base.py` for "to"
and raise if no installation is found so that we don't have to check anywhere else. Afaik, the imports in python is done in a "lazy" manner as such this should work. 
@bharatkumarreddy could you confirm if this is working? (test without tf installation too)